packages/lesswrong/components/ea-forum/giving-portal/ElectionCandidatesList.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useState } from "react";
+import { Components, registerComponent } from "../../../lib/vulcan-lib";
+import { useElectionCandidates } from "./hooks";
+import { isElectionCandidateSort } from "../../../lib/collections/electionCandidates/views";
+import type { SettingsOption } from "../../../lib/collections/posts/dropdownOptions";
+import classNames from "classnames";
+
+const styles = (theme: ThemeType) => ({
+  root: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "flex-start",
+    gap: "16px",
+  },
+  grid: {
+    display: "flex",
+    flexWrap: "wrap",
+    gap: "16px",
+    rowGap: "12px",
+  },
+  dropdown: {
+    "& .ForumDropdownMultiselect-button": {
+      color: theme.palette.givingPortal[0],
+      backgroundColor: theme.palette.givingPortal[500],
+      "&:hover": {
+        backgroundColor: theme.palette.grey[250],
+        color: theme.palette.grey[1000],
+      },
+    },
+  },
+});
+
+const sortOptions: Record<ElectionCandidatesSort, SettingsOption> = {
+  mostPreVoted: {
+    label: "Most pre-voted",
+  },
+  name: {
+    label: "Name A-Z",
+  },
+  recentlyAdded: {
+    label: "Recently added",
+  },
+};
+
+const ElectionCandidatesList = ({className, classes}: {
+  className?: string,
+  classes: ClassesType,
+}) => {
+  const [sortBy, setSortBy] = useState<ElectionCandidatesSort>("mostPreVoted");
+  const {results, loading} = useElectionCandidates(sortBy);
```

Comment:
I noticed that changing my pre-votes doesn't change the ordering (which is good), but probably ideally if I change the sort and then change it back to the default sort, it should correctly be ordered, instead of reverting back (like it currently does). But I think this isn't a big deal so feel free to ignore.


https://github.com/ForumMagnum/ForumMagnum/assets/9057804/0e3a5424-e2ac-409f-bcdb-37820adaa594

Reply:
Makes sense. Given time constraints I'm going to ignore for now, but I'll keep it in mind if we have some spare time later.